pkg/metrics/prometheus.go

```diff
@@ -0,0 +1,152 @@
+package metrics
+
+import (
+	"github.com/grafana/grafana/pkg/log"
+	"github.com/prometheus/client_golang/prometheus"
+	"net/http"
+	"strings"
+)
+
+const (
+	namespace = "grafana"
+	subsystem = "backend"
+)
+
+type PrometheusPublisher struct {
+	CounterCollectors map[string]*prometheus.CounterVec
+	GaugeCollectors   map[string]*prometheus.GaugeVec
+}
+
+func CreatePrometheusPublisher() (*PrometheusPublisher, error) {
+	go func() {
+		http.Handle("/metrics", prometheus.Handler())
+		err := http.ListenAndServe(":9200", nil)
```

Comment:
> We think of that in terms of average logins per second.
But won't that make totals over say a day very inaccurate (especially with server restarts, etc)?  or if the rate varies a lot. 

> It's syntactic sugar over rate, it multiplies the rate by the time range.

Which sounds like a very bad thing if you want to sum events. But maybe it becomes more accurate than I think due to the way things are stored in Prometheus? 

It's an interesting discussion. I definitely seeing some pros to storing raw values, but to me storing deltas makes it soo much more easier to reason about the metrics, transform, divide, sum them etc. And having roll ups to do long term trending is what I really like about TSDBs as well so that's why I have always liked deltas more. But looks like Prometheus makes working with raw counters a bit easier than say Graphite or InfluxDB (where they have been an incredible source of pain and confusion, especially for InfluxDB).